src/entities/product/validation/createProductValidation.schema.ts

```diff
@@ -0,0 +1,33 @@
+import * as yup from 'yup';
+import {validationErrors} from '@app/constants/validationErrors.constant';
+import {CategoriesEnum} from '@app/constants/categories.enum';
+import {SubcategoriesEnum} from '@app/constants/subcategories.enum';
+import {SellerEnum, RatingEnum, StateEnum} from '@app/constants/product.enum';
+import {validationConstants} from '@app/constants/validation.constant';
+
+const {MIN_LENGTH_TITLE, MIN_LENGTH_DESCRIPTION} = validationConstants;
+const {FIELD_REQUIRED, INVALID_VALUE, INVALID_MIN_LENGTH} = validationErrors;
+
+export const createProductSchema = yup.object().shape({
+  title: yup.string().min(MIN_LENGTH_TITLE, INVALID_MIN_LENGTH(MIN_LENGTH_TITLE)).required(FIELD_REQUIRED),
+  categoryId: yup.number().oneOf(Object.values(CategoriesEnum), INVALID_VALUE).required(FIELD_REQUIRED),
+  subcategoryId: yup.number().oneOf(Object.values(SubcategoriesEnum), INVALID_VALUE).required(FIELD_REQUIRED),
+  description: yup
+    .string()
+    .min(MIN_LENGTH_DESCRIPTION, INVALID_MIN_LENGTH(MIN_LENGTH_DESCRIPTION))
+    .required(FIELD_REQUIRED),
+  detailedDescription: yup.array().of(
+    yup.object().shape({
+      label: yup.string().required(FIELD_REQUIRED),
+      value: yup.string().required(FIELD_REQUIRED)
+    })
+  ),
+  images: yup.array().of(yup.string()).required(FIELD_REQUIRED),
+  price: yup.number().required(FIELD_REQUIRED),
+  quantity: yup.number().required(FIELD_REQUIRED),
```

Comment:
It could. Now I added this for price, discount, quantity and rating:
```
price: yup.number().positive().required(FIELD_REQUIRED),
 @Prop({required: true, min: 0})
  price: number;
```